File: VC/1.0.0/service/project/Care4Toolbox/src/main/java/com/care4u/toolbox/tool/ToolDto.java
package com.care4u.toolbox.tool;

import com.care4u.toolbox.group.sub_group.SubGroupDto;

import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public class ToolDto {
	
	private long id;
	
	private String name;
	
	private SubGroupDto subGroupDto;
	
	private String code;
	
	private String buyCode;
	
	private String engName;
	
	private String spec;
	
	private String unit;
	
	private int price;
	
	private int replacementCycle;
	
	@Builder
	public ToolDto(long id, String name, SubGroupDto subGroupDto, String code, String buyCode, String engName, 
			String spec, String unit, int price, int replacementCycle) {
		this.id = id;
		this.name = name;
		this.subGroupDto = subGroupDto;
		this.code = code;
		this.buyCode = buyCode;
		this.engName = engName;
		this.spec = spec;
		this.unit = unit;
		this.price = price;
		this.replacementCycle = replacementCycle;
	}
	
	public ToolDto(Tool tool) {
		this.id = tool.getId();
		this.name = tool.getName();
		this.subGroupDto = new SubGroupDto(tool.getSubGroup());
		this.code = tool.getCode();
		this.buyCode = tool.getBuyCode();
		this.engName = tool.getEngName();
		this.spec = tool.getSpec();
		this.unit = tool.getUnit();
		this.price = tool.getPrice();
		this.replacementCycle = tool.getReplacementCycle();
	}
	
}

File: App/BTClient/app/src/main/java/com/mrsmart/standard/tool/ToolDto.kt
package com.mrsmart.standard.tool

import com.mrsmart.standard.tool.group.SubGroupDto

data class ToolDto(
    val id: Long,
    val subGroupDto: SubGroupDto,
    val code: String,
    val name: String,
    val engName: String,
    val spec: String,
    val unit: String,
    val price: Int,
    val replacementCycle: Int,
    val buyCode: String
) {
    fun toToolDtoSQLite (): ToolSQLite {
        val buyCode = buyCode ?: ""
        return ToolSQLite(id, subGroupDto.name, subGroupDto.mainGroupDto.name, code, name, engName, spec, unit, price, replacementCycle, buyCode)
    }
}
